src/main/java/com/example/onlinebookstore/controller/CategoriesController.java

```diff
@@ -0,0 +1,86 @@
+package com.example.onlinebookstore.controller;
+
+import com.example.onlinebookstore.dto.book.BookDtoWithoutCategoryIds;
+import com.example.onlinebookstore.dto.category.CategoryRequestDto;
+import com.example.onlinebookstore.dto.category.CategoryResponseDto;
+import com.example.onlinebookstore.service.BookService;
+import com.example.onlinebookstore.service.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Positive;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Categories management", description = "Endpoints for managing books categories")
+@RestController
+@RequestMapping("/api/categories")
+@RequiredArgsConstructor
+@Validated
+public class CategoriesController {
+    private final BookService bookService;
+    private final CategoryService categoryService;
+
+    @PreAuthorize("hasRole('ROLE_USER') || hasRole('ROLE_ADMIN')")
```

Comment:
you are setting the default role USER to every user, so checking hasRole USER is redundant